### ISSUE TITLE
PC algorithm logic

### ISSUE THREAD

**Issue by kenneth-lee-ch** [py-why/causal-learn#129]
In the `causal-learn/causallearn/utils/PCUtils/SkeletonDiscovery.py`, why does it need to add `sepsets` to `cp.sepset` at lines 123-124? If the p-value return at line 101 is not greater than alpha for all `S` in line 100, doesn't that mean you force an empty set to be a separating set?

**Comment by kunwuz** [comment]
Yes, and it is consistent with the pseudo-code of PC, where an empty set could be a separating set.

**Comment by kenneth-lee-ch** [comment]
I mean you will infer d-separation with respect to an empty set regardless of the test result.

**Comment by kunwuz** [comment]
Thanks for the explanation. @zhi-yi-huang tried to fix this issue in #161 (thanks so much!). Please let us know if there are any questions.
